spectrocloud/resource_cluster_tke.go

```diff
@@ -332,25 +332,28 @@ func resourceClusterTkeUpdate(ctx context.Context, d *schema.ResourceData, m int
 
 		for _, mp := range ns {
 			machinePoolResource := mp.(map[string]interface{})
-			name := machinePoolResource["name"].(string)
-			hash := resourceMachinePoolTkeHash(machinePoolResource)
-
-			machinePool := toMachinePoolTke(machinePoolResource)
-
-			var err error
-			if oldMachinePool, ok := osMap[name]; !ok {
-				log.Printf("Create machine pool %s", name)
-				err = c.CreateMachinePoolTke(cloudConfigId, machinePool)
-			} else if hash != resourceMachinePoolTkeHash(oldMachinePool) {
-				log.Printf("Change in machine pool %s", name)
-				err = c.UpdateMachinePoolTke(cloudConfigId, machinePool)
+			if machinePoolResource["name"].(string) != "" {
```

Comment:
```suggestion
                        // since known issue in TF SDK: https://github.com/hashicorp/terraform-plugin-sdk/issues/588
			if machinePoolResource["name"].(string) != "" {
```